src/nlp/meta.jl

```diff
@@ -117,10 +117,10 @@ struct NLPModelMeta <: AbstractNLPModelMeta
   name::String
 
   function NLPModelMeta(
-    nvar;
-    x0 = zeros(nvar),
-    lvar = -Inf * ones(nvar),
-    uvar = Inf * ones(nvar),
+    nvar::Int;
+    x0::AbstractVector{T} = zeros(nvar),
```

Comment:
I think it should be `zeros(T, nvar)` and similar for `lvar` and `uvar` here.

Reply:
There is no inference of type `T` because `x0` is a keyword argument. It would need to be `NLPModelMeta{T}`, or either `T` or `x0` would need to be a mandatory argument. That's why I suggested `NLPModelMeta{T,S}` and a constructor with `x0` as mandatory above. What do you think?

Reply:
Ohhh, ok sorry for that.

Completely agree with you with the `x0` as a mandatory argument.